cmd/asset-syncer/postgresql_utils_test.go

```diff
@@ -124,11 +124,11 @@ func Test_PGfilesExist(t *testing.T) {
 	pgManager := &postgresAssetManager{man}
 	m.On(
 		"Query",
-		`SELECT * FROM files WHERE info ->> 'ID' = $1 AND info ->> 'digest' = $2`,
+		`SELECT * FROM files WHERE chart_files_id = $1 AND info ->> 'Digest' = $2`,
 		[]interface{}{id, digest},
 	)
 	exists := pgManager.filesExist(id, digest)
-	if exists != true {
+	if exists != false {
```

Comment:
The name of this test implies it's testing that it can (successfully) verify that a file exists, but the switch in value here implies it will now call `t.Errorf` if the file exists? It's pretty confusing (imo). The current logic as it is with this change (even though I can't make semantic sense out of it) could be just:
```
if pgManager.filesExist(id, digest) {
    t.Errorf("Failed to check if file exists")
}
```

Reply:
The goal of the name of the test is to say that it's testing the method `fileExists` (which may appear to say that the file has to exist). The change in the result is caused by the change that we only mark it as valid if the returned rows has at least lenght 1. Before we were just checking if the query returned an error to assume that in that case the file exist.

That's why the result of this test changed. The error keeps being `nil` but the mock query is not returning any row.

It's true that the test would be more valuable if we are actually be able to verify if a file exists. We can do that with the library I am using in #1394 which at least lets you return rows. I will do so once I merge that PR in this branch.

Reply:
Thanks, that last commit makes the test much clearer.